packages/angular/src/errorhandler.ts

```diff
@@ -50,6 +50,35 @@ function extractHttpModuleError(error: HttpErrorResponse): string | Error {
   return error.message;
 }
 
+function isObject(value: unknown): value is Record<string, unknown> {
+  return value !== null && typeof value === 'object';
+}
+
+function hasOwnProperty<ObjectType extends Record<string, unknown>, PropertyType extends PropertyKey>(
+  object: ObjectType,
+  propertyName: PropertyType,
+): object is ObjectType & Record<PropertyType, unknown> {
+  return Object.prototype.hasOwnProperty.call(object, propertyName);
+}
+
+function hasErrorName(value: Record<string, unknown>): value is Pick<Error, 'name'> {
+  return hasOwnProperty(value, 'name') && typeof value.name === 'string';
```

Comment:
Not saying it's wrong but why do we need the `hasOwnProperty` check here? Wouldn't it be enough to check for `typeof value.name === 'string'`? 

Or actually, for the string check, we could simply reuse our `isString` check from `@sentry/utils`.